src/datasources/system/types.ts

```diff
@@ -6,7 +6,8 @@ export enum QueryType {
 }
 
 export interface SystemQuery extends DataQuery {
-  queryKind: QueryType
+  queryKind: QueryType,
+  systemId: string
```

Comment:
Similar to my other comment, since this field can be either the alias or the id, we should name it more generically.